### DISCUSSION TITLE
puppet modules required on server

### DISCUSSION THREAD

**Discussion by dom0317** [choria-io/general#48] [General]
Hi,

On a server on which we are never going to initiate any mco commands is all the choria software required to be installed?
It will need the choria rpm, the mcorpc and nats gems and should have the choria-choria puppet module but are the other puppet modules really needed. Why would the choria-mcollective_agent_puppet and other agent puppet modules be installed if we are not going to run mco commands on that server? And without any mcollective plugins are the choria-mcollective and choria-mcollective_choria modules really required?

**Comment by ripienaar** [comment]
If you dont set the client=true boolean the modules will not install the things that servers dont need.

A agent is broken into 2 or 3 parts:

 1. The schema for the agent - the DDL we call it
 2. The logic that gets executed when the agent is called
 3. An optional client like `mco puppet` for doing deeper, often multi step, interactions with the agent

1 goes everywhere.  2 only on servers. 3 only on clients. 

`choria-mcollective_agent_puppet` holds all 3 and will do the right split based on `client=true` and `server=true`
